core/src/main/scala/kafka/server/metadata/BrokerMetadataListener.scala

```diff
@@ -325,7 +325,7 @@ class BrokerMetadataListener(
     try {
       _image = _delta.apply()
     } catch {
-      case t: Throwable => metadataLoadingFaultHandler.handleFault(s"Error applying metadata delta $delta", t)
+      case t: Throwable => throw metadataLoadingFaultHandler.handleFault(s"Error applying metadata delta $delta", t)
```

Comment:
What is your expectation if this throws? I ask this because at this point `_delta` and `_image` are not in a coherent or consistent state.

Reply:
I think `_image` and `_delta` will be unchanged if the apply call throws. The publish will raise an exception back to the StartPublishingEvent, HandleCommitEvent, or HandleSnapshotEvent. We should not see the `_image` get updated (unless of course a future snapshot could apply without error)

Reply:
@mumrah and I spoke offline about this. How about documenting the state of the broker at this point?

Reply:
Looking at the documentation for `FaultHandler` it looks like it is allowed for `handleFault` to return `null`.

Reply:
Hm yea, does look we can return `null`. In this case I believe we will see a `NullPointerException`. The EventQueue should still keep going in this case